test/validation/api/ipsec/ipsec.c

```diff
@@ -763,6 +763,35 @@ static void ipsec_pkt_auth_err_set(odp_packet_t pkt)
 	odp_packet_copy_from_mem(pkt, len - sizeof(data), sizeof(data), &data);
 }
 
+static void ipsec_pkt_v4_check_udp_encap(odp_packet_t pkt)
+{
+	uint32_t l3_off = odp_packet_l3_offset(pkt);
+	odph_ipv4hdr_t ip;
+
+	odp_packet_copy_to_mem(pkt, l3_off, sizeof(ip), &ip);
+
+	CU_ASSERT(ip.proto == ODPH_IPPROTO_UDP);
+}
+
+static void ipsec_pkt_v6_check_udp_encap(odp_packet_t pkt)
+{
+	uint32_t l3_off = odp_packet_l3_offset(pkt);
+	odph_ipv6hdr_ext_t ext;
+	odph_ipv6hdr_t ip;
+	uint8_t next_hdr;
+
+	odp_packet_copy_to_mem(pkt, l3_off, sizeof(ip), &ip);
+
+	next_hdr = ip.next_hdr;
+
+	if (ip.next_hdr == ODPH_IPPROTO_HOPOPTS) {
+		odp_packet_copy_to_mem(pkt, l3_off + sizeof(ip), sizeof(ext), &ext);
+		next_hdr = ext.next_hdr;
```

Comment:
I wonder why this handling of hop-by-hop options is here? AFAICS, we do not have that kind of packets in the tests so this branch would never be executed. Why not simply remove it and fail the validation test if there are any extension headers?

Reply:
When I tried IPv6 with transport mode, the test was failing. Hence added this. Basically, the test vector used has hop-by-hop options and in transport mode, ESP additions would be after all IPv6 headers.

Reply:
You are right. Now I can see the hop-by-hop options in the test vectors.